railties/lib/rails/test_unit/minitest_plugin.rb

```diff
@@ -60,7 +60,7 @@ def self.plugin_rails_options(opts, options)
   # Running several Rake tasks in a single command would trip up the runner,
   # as the patterns would also contain the other Rake tasks.
   def self.rake_run(patterns) # :nodoc:
-    @rake_patterns = patterns
+    (@rake_patterns ||= []).push(*patterns)
```

Comment:
The comment is still valid with `@rake_patterns` - it's targeted at fixing `rake db:migrate test` rather than two test tasks. Without it, the plugin would run tests listed in the argument list, which is `['db:migrate' ,'test']` - instead `rake_run` stores the paths from test:\* tasks so running test + non-test tasks works.

This was originally added in df744b5, but missing a test to cover it and the conditional above, so I added one in b5a9428.

I think you're right that `rake_run` (or rather `@rake_patterns`) is a bit obsolete now that `autorun` is used. The test files can be required immediately and `autorun` will handle it. I've removed `@rake_patterns` in a12803d, but kept `rake_run` as it makes the tasks in railties/lib/rails/test_unit/testing.rake a little less repetitive.